glean-core/ios/Glean/Dispatchers.swift

```diff
@@ -88,17 +82,72 @@ class Dispatchers {
-            } else {
-                operationQueue.addOperation(operation)
+                serialOperationQueue.waitUntilAllOperationsAreFinished()
             }
         }
+    }
+
+    /// This function launches a background `Operation` on a concurrently executed queue.
+    ///
+    /// - parameters:
+    ///   * operation: The `Operation` to execute
+    ///
+    /// This function is used to execute tasks in an asynchrounous manner and still give us the ability
+    /// to cancel the tasks by creating them as `Operation`s rather than using GCD.
+    ///
+    /// This function specifically ignores the `queueInitialTasks` flag because the only tasks that
+    /// should be launched by this are the ping upload schedulers and those should run regardless of
+    /// the initialized state.
+    ///
+    /// If `testingMode` is enabled, then launch will just execute the task rather than adding it to the
+    /// concurrent queue to avoid asynchronous issues while testing
+    func launchConcurrent(operation: Operation) {
+        if testingMode {
+            operation.start()
+            operation.waitUntilFinished()
+        } else {
+            concurrentOperationsQueue.addOperation(operation)
+        }
+    }
 
-        return operation
+    /// This function launches a background block of code on a concurrently executed queue.
+    ///
+    /// - parameters:
+    ///   * block: The block of code to execute, as a Closure that accepts no arugments and returns Void
+    ///
+    /// This function is used to execute tasks in an asynchrounous manner and still give us the ability
+    /// to cancel the tasks by creating them as `Operation`s rather than using GCD.
+    ///
+    /// This function specifically ignores the `queueInitialTasks` flag because the only tasks that
+    /// should be launched by this are the ping upload schedulers and those should run regardless of
+    /// the initialized state.
+    ///
+    /// If `testingMode` is enabled, then launch will just execute the task rather than adding it to the
+    /// concurrent queue to avoid asynchronous issues while testing
+    func launchConcurrent(block: @escaping () -> Void) {
+        launchConcurrent(operation: BlockOperation(block: block))
+    }
+
+    /// Cancel any pending background tasks
+    func cancelBackgroundTasks() {
+        // This will remove all queued operations prior to Glean.initialize() so that they won't be
+        // executed if flushQueuedInitialTasks() is called
+        preInitOperations.removeAll()
+
+        // This will cancel operations in the serially executed queue. This includes most of the
+        // metrics recording and things that need to execute in order. This would not stop the currently
+        // executing operation, but would prevent all remaining operations from executing.
+        serialOperationQueue.cancelAllOperations()
+
+        // This will cancel operations that are executing concurrently. This doesn't abort the thread
```

Comment:
```suggestion
        // This will cancel operations that are executing concurrently. 
        // This doesn't abort running operations immediately
```